src/app/[game]/SearchBar.tsx

```
@@ -0,0 +1,114 @@
+
+    const timeoutRef = React.useRef<ReturnType<typeof setTimeout> | null>(null);
+
+    const searchProviderLoadedPromiseRef = React.useRef<{resolve?:(res:SearchContextLoaded)=>void,promise: Promise<SearchContextLoaded>}>(null);
+    const isLoading = searchProvider.LOADING_FROM_SSR || searchProvider.DEVELOPMENT__LOADING_HASH;
+    searchProviderLoadedPromiseRef.current ??= (()=>{
+        if (!isLoading) return {promise: Promise.resolve(searchProvider as SearchContextLoaded)};
+        let hoistedResolveF: (res:SearchContextLoaded)=>void;
+        const promise = new Promise<SearchContextLoaded>((resolve) => {
+            hoistedResolveF = (loaded)=> {
+                resolve(loaded);
+                delete searchProviderLoadedPromiseRef.current!.resolve;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)